src/watchmaker/utils/imds/detect/providers/azure_provider.py

```diff
@@ -47,18 +47,24 @@ def check_vendor_file(self):
 
         Read file /sys/class/dmi/id/sys_vendor
         """
-        self.logger.debug('Checking Azure vendor file')
+        self.logger.debug("Checking Azure vendor file")
 
         # Convert a local vendor file to a URI
-        if exists(self.vendor_file):
-            vendor_file_path = utils.uri_from_filepath(self.vendor_file)
-            try:
-                check_vendor_file = utils.urlopen_retry(
-                    vendor_file_path, self.DEFAULT_TIMEOUT)
-                if "Microsoft Corporation" in check_vendor_file.read():
-                    return True
-            except (ValueError, utils.urllib.error.URLError):
-                pass
+        if not exists(self.vendor_file):
+            self.logger.debug("Vendor file does not exist")
+            return False
+
+        self.logger.debug("Vendor file exists")
+        vendor_file_path = utils.uri_from_filepath(self.vendor_file)
+        try:
+            check_vendor_file = utils.urlopen_retry(
+                vendor_file_path, self.DEFAULT_TIMEOUT
+            )
+            if b"Microsoft Corporation" in check_vendor_file.read():
```

Comment:
Do you happen to remember why this needs to be `b""` now?

Reply:
The read method returns bytes, I just updated the aws one to convert the strnig to bytes, tested it, and it worked.  The aws code path always passed on metadata server and never checked the file, had to force the test.  When we check for azure the detect code checked the file before the metadata server and since that file exists on linux, regardless of AWS or Azure, we did hit the bytes like error not str, so I added the bytes conversion.